lib/Doctrine/ORM/PersistentCollection.php

```diff
@@ -187,13 +200,8 @@ public function hydrateSet($key, $element)
     {
         $this->collection->set($key, $element);
 
-        // If _backRefFieldName is set, then the association is bidirectional
-        // and we need to set the back reference.
         if ($this->backRefFieldName && $this->association['type'] === ClassMetadata::ONE_TO_MANY) {
-            // Set back reference to owner
-            $this->typeClass->reflFields[$this->backRefFieldName]->setValue(
-                $element, $this->owner
-            );
+            $this->completeBidirectionalAssociationLink($element);
```

Comment:
No tests failed when I added this here, now it's also executing this (but I'm not sure if it should):

```php
$this->em->getUnitOfWork()->setOriginalEntityProperty(
    spl_object_hash($element),
    $this->backRefFieldName,
    $this->owner
);
```